File: js/shapes/shape.js
class Shape {
  constructor(type, borderThickness = 0) {
    this.type = type;
    this.borderThickness = borderThickness;
  }

  /**
   *
   * @param {*} borderThickness
   * @returns Area of the shape
   * This should override in each shapes
   */
  calArea(borderThickness = 0) {
    return borderThickness * borderThickness;
  }

  getAreaWithBorder() {
    return this.calArea(this.borderThickness);
  }

  getAreaWithoutBorder() {
    return this.calArea();
  }

  getAreaOfBorder() {
    return this.calArea(this.borderThickness) - this.calArea();
  }
}

module.exports = Shape;

File: c#/shapes/Shape.cs
namespace Canvas;

public class Shape
{
    public string? Type { get; set; }
    public int BorderThickness { get; set; }

    public virtual int CalArea(int borderThickness)
    {
        return borderThickness * borderThickness;
    }

    public int GetAreaWithBorder()
    {
        return CalArea(BorderThickness);
    }

    public int GetAreaWithoutBorder()
    {
        return CalArea(0);
    }

    public int GetAreaOfBorder()
    {
        return GetAreaWithBorder() - GetAreaWithoutBorder();
    }
}
